src/business-logic/services/test/device-service.spec.ts

```diff
@@ -0,0 +1,41 @@
+import { DeviceServiceImplementation } from '../device-service-implementation'
+import { Device, TelemetricsDevice } from '../../../model/entities/device'
+import { mock, when, verify, instance } from '@typestrong/ts-mockito'
+import { MongoDeviceRepository } from '../../../data-access/repositories/mongo/mongo-device-repository'
+import { DeviceService } from '../interfaces/device-service'
+import { DeviceEventEmitter } from '../interfaces/device-event-emitter'
+
+describe('DeviceService', () => {
+  let testDevice: TelemetricsDevice
+  let deviceRepository: MongoDeviceRepository
+  let deviceEventEmitter: DeviceEventEmitter
+  const mockedValue = mock<MongoDeviceRepository>() 
+
+  beforeEach(() => {
+    testDevice = mock<TelemetricsDevice>()
+    deviceRepository = instance(mockedValue)
+    deviceEventEmitter = mock<DeviceEventEmitter>()
```

Comment:
This is setting up "global" mock for all tests. We try to avoid that and instead setup the mock for each test individually.
The reason for this is, if you change this mock because you need a slight difference in a new test, you now have to verify that _every single test_ still works as intented. This can get quite tedious when there is a lot of tests.

What we do for common mock shared between tests is to have helper methods create it, that the tests can call.